### ISSUE TITLE
Regression: Zero-length delimited identifier is generated for enum columns when using <outputSchemaToDefault/>

### ISSUE THREAD

**Issue by baev** [jOOQ/jOOQ#13723]
### Expected behavior

We have a few test failures after 3.17.0 upgrade. 

```
jOOQ; bad SQL grammar [insert into "account_authority" ("account_id", "authority") values (?, cast(? as ""."authority_type"))]; nested exception is org.postgresql.util.PSQLException: ERROR: zero-length delimited identifier at or near """"
  Position: 84
```

The generated columns:

```
 /**
     * The column <code>account_authority.account_id</code>.
     */
    public final TableField<AccountAuthorityRecord, Long> ACCOUNT_ID = createField(DSL.name("account_id"), SQLDataType.BIGINT.nullable(false), this, "");

    /**
     * The column <code>account_authority.authority</code>.
     */
    public final TableField<AccountAuthorityRecord, AuthorityType> AUTHORITY = createField(DSL.name("authority"), SQLDataType.VARCHAR.nullable(false).asEnumDataType(io.qameta.allure.testops.uaa.db.enums.AuthorityType.class), this, "");
```

### Actual behavior

No exception occurs

### Versions

- jOOQ: 3.17.0 (pro, java-11 distro)
- Java: Java 17
- Database (include vendor): PostgreSQL 14
- OS: MacOS m1
- JDBC Driver (include name if inofficial driver): PostgreSQL latest

**Comment by lukaseder** [comment]
Thanks for your report. Do you have schema mapping in place, or why is the schema omitted?

**Comment by baev** [comment]
there is nothing special:

```
create type authority_type as enum (
     'SOME','OTHER'
);

create table account (
    id bigserial primary key,
    username text not null
);

create table account_authority (
    account_id bigint not null references account on update cascade on delete cascade,
    authority authority_type not null,
    primary key (account_id, authority)
);
```

no custom mappings are specified

**Comment by baev** [comment]
and the method that fails:

```java
public void addAccountAuthorities(final AccountRecord ar,
                                      final AuthorityType first,
                                      final AuthorityType... others) {
        final List<Row2<Long, AuthorityType>> rows = Stream.concat(Stream.of(first), Stream.of(others))
                .map(authr -> DSL.row(ar.getId(), authr))
                .collect(Collectors.toList());

        dsl.insertInto(ACCOUNT_AUTHORITY)
                .columns(ACCOUNT_AUTHORITY.ACCOUNT_ID, ACCOUNT_AUTHORITY.AUTHORITY)
                .valuesOfRows(rows)
                .execute();
    }
```

**Comment by lukaseder** [comment]
OK, thanks for following up. So, let's try to reproduce...

**Comment by lukaseder** [comment]
No, sorry, I can't reproduce it yet. In my case, the enum is part of the `"public"` schema, and the query produces the correct result:

```sql
insert into "public"."t_book" ("id", "author_id", "published_in", "status")
values
  (?, ?, ?, cast(? as "public"."u_book_status")), 
  (?, ?, ?, cast(? as "public"."u_book_status"))
```

If you can provide more info related to the lack of schema qualification, that would be great. I'll try to reproduce the issue on my side thinking about what could cause it.

**Comment by lukaseder** [comment]
OK, I guess key here is to set `<outputSchemaToDefault>true</outputSchemaToDefault>`. Now I'm getting a query like this, as well:

```sql
insert into "t_book" ("id", "author_id", "published_in", "status")
values (?, ?, ?, cast(? as ""."u_book_status"))
```

**Comment by baev** [comment]
yeah, I do confirm we are using `outputSchemaToDefault` set to true

**Comment by lukaseder** [comment]
Thanks for the confirmation. I can confirm this is a regression from 3.16 -> 3.17

**Comment by lukaseder** [comment]
Probably introduced by #13534

**Comment by lukaseder** [comment]
Nope. https://github.com/jOOQ/jOOQ/issues/13035 introduced this.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0 and 3.17.1 (https://github.com/jOOQ/jOOQ/issues/13724). Thanks again for your report!

**Comment by lukaseder** [comment]
Note, this regression failed for enum types only and not for tables, because for tables, there's a redundant check in `TableImpl` (and other types):

```java
if (mappedSchema != null && !"".equals(mappedSchema.getName())) { ...
```

We have integration tests for `<outputSchemaToDefault/>` when using tables, but we didn't have integration tests for that feature when using enums.
